bbswitch.c

```diff
@@ -260,6 +260,7 @@ static void bbswitch_off(void) {
     pci_save_state(dis_dev);
     pci_clear_master(dis_dev);
     pci_disable_device(dis_dev);
+#if defined(CONFIG_X86) || defined(CONFIG_ARM64)
```

Comment:
acpi_bus_get_device is provided by scan.c in the kernel which is compiled whenever

```
obj-$(CONFIG_ACPI)              += acpi.o \
                                        acpica/
acpi-y                         += scan.o
```
Wouldn't it be better to ifdef based not on the architecture, but on `#ifdef CONFIG_ACPI` ?

Which seems to be the more common pattern across the kernel drivers to do more things if ACPI is available, in an arch independent way. For example, in case ACPI is added to more arches in the future.
```suggestion
#if defined(CONFIG_X86) || defined(CONFIG_ARM64)
```
```suggestion
#if defined(CONFIG_X86) || defined(CONFIG_ARM64)
```
```suggestion
#ifdef CONFIG_ACPI
```

Reply:
That makes sense, I like to rely on CONFIG_ACPI more! Thanks for looking at this!